File: src/main/java/com/example/the_machine/dto/content/ToolCallContentDto.java
package com.example.the_machine.dto.content;

import com.fasterxml.jackson.databind.JsonNode;

public record ToolCallContentDto(
    String name,
    JsonNode arguments
) implements ContentBlockDto {

  @Override
  public String type() {
    return ContentBlockTypes.TOOL_CALL;
  }
}

File: src/main/kotlin/com/example/the_machine/dto/content/ToolCallContentDto.kt
package com.example.the_machine.dto.content

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

@Serializable
@SerialName(ContentBlockTypes.TOOL_CALL)
data class ToolCallContentDto(
  val name: String,
  val arguments: JsonElement
) : ContentBlockDto